src/Renderer/ThreeExtended/PntsLoader.js

```diff
@@ -0,0 +1,113 @@
+import * as THREE from 'three';
+
+export default {
+    parse: function parse(buffer, layer) {
+        if (!buffer) {
+            throw new Error('No array buffer provided.');
+        }
+
+        const array = new Uint8Array(buffer);
+        const view = new DataView(buffer);
+
+        let byteOffset = 0;
+        const pntsHeader = {};
+
+        // Magic type is unsigned char [4]
+        pntsHeader.magic = decodeFromCharCode(array.subarray(byteOffset, 4));
+        byteOffset += 4;
+
+        if (pntsHeader.magic) {
+            // Version, byteLength, batchTableJSONByteLength, batchTableBinaryByteLength and batchTable types are uint32
+            pntsHeader.version = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.byteLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.FTJSONLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.FTBinaryLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.BTJSONLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.BTBinaryLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            // binary table
+            if (pntsHeader.FTBinaryLength > 0) {
+                // readTestFT(array, byteOffset, pntsHeader.FTJSONLength);
+                return parseFeatureBinary(array, byteOffset, pntsHeader.FTJSONLength, buffer, layer);
+            }
+
+            // batch table
+          /*
+            if (pntsHeader.BTBinaryLength > 0) {
+                console.log('BTB');
+            }
+            */
+        } else {
+            throw new Error('Invalid pnts file.');
+        }
+    },
+};
+
+function parseFeatureBinary(array, byteOffset, FTJSONLength, buffer, layer) {
+    const subArrayJson = decodeFromCharCode(array.subarray(byteOffset, FTJSONLength + byteOffset));
+    const parseJSON = JSON.parse(subArrayJson);
+
+    const view = new Float32Array(buffer);
```

Comment:
This is doing a full copy of buffer.
You should pass the arrayBuffer to this function and create directly the positionArray below:
```
const positionArray = new Float32Array(buffer,  byteOffsetPos, lengthFeature * 3);
```
(and the same is true for `colorArray`)